src/game/Protocol/WorldSocket.cpp

```diff
@@ -308,11 +413,11 @@ int WorldSocket::HandleAuthSession(WorldPacket& recvPacket)
     else
     {
         sLog.Out(LOG_BASIC, LOG_LVL_ERROR, "WorldSocket::HandleAuthSession: Unrecognized Platform '%s' for account '%s' from %s", platform.c_str(), account.c_str(), address.c_str());
-        return -1;
+        return HandlerResult::Fail;
     }
 
-    // NOTE ATM the socket is single-threaded, have this in mind ...
-    ACE_NEW_RETURN(m_Session, WorldSession(id, this, AccountTypes(security), mutetime, locale), -1);
+    // TODO: cMangos has session->RequestNewSocket() and can attach to an exising session
```

Comment:
In vmangos a new session is actually a new session. If you disconnect and reconnect, it should not reuse the old session, but create a new one. We don't need to change this.